src/pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj

```diff
@@ -80,6 +80,43 @@
                               PreferredPackages="$(Id);@(RuntimeDependency)" />
   </Target>
 
+  <UsingTask TaskName="GetVersionFromAssembly" AssemblyFile="$(LocalBuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.Local.dll"/>
+
+  <Target Name="GenerateHashVersionsFile" BeforeTargets="GetPackageFiles">
+
+    <PropertyGroup>
+      <_SystemRuntimePath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.runtime.dll'))">%(File.Identity)</_SystemRuntimePath>
+      <_SystemPrivateCoreLibPath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.private.corelib.dll'))">%(File.Identity)</_SystemPrivateCoreLibPath>
+    </PropertyGroup>
+
+    <GetVersionFromAssembly PathToAssembly="$(_SystemRuntimePath)">
+      <Output TaskParameter="AssemblyHash" PropertyName="CoreFxHash" />
```

Comment:
(Just a comment) CoreFX packages do have a version.txt file inside with the commit hash, but I don't know if it's reasonable to read that file from here. (CoreCLR doesn't have version.txt, anyway.)